### ISSUE TITLE
Error exit(may relate to sync)

### ISSUE THREAD

**Issue by kis87988** [botorange/wechaty-puppet-padpro#9]
[syncIssue112718.log](https://github.com/botorange/wechaty-puppet-padpro/files/2623435/syncIssue112718.log)
some extend log
```log
00:36:03 INFO Bot finis(99,uncaughtException,Error: connect ETIMEDOUT 203.205.129.101:80)
00:36:03 INFO Bot finis() broadcast quiting message for bot
00:36:03 VERB Wechaty say(Wechaty will exit 99 because of uncaughtException )
00:36:03 VERB Puppet selfId()
00:36:03 VERB Contact say(Wechaty will exit 99 because of uncaughtException )
00:36:03 VERB PuppetPadpro messageSend({"contactId":"wxid_mask"}, Wechaty will exit 99 because of uncaughtException )
00:36:03 SILL PadproGrpc GrpcSendMessage(wxid_mask, Wechaty will exit 99 because of uncaughtException , undefined)
00:36:08 SILL PadproManager throttleQueue emit heartbeat.
00:36:08 SILL PuppetPadpro startWatchdog() padproManager.on(heartbeat)
00:36:08 SILL Wechaty memoryCheck() free: 232 MB, require: 4 MB
Error: connect ETIMEDOUT 203.205.129.101:80
    at TCPConnectWrap.afterConnect [as oncomplete] (net.js:1113:14)
00:36:08 INFO Bot finis(1,exit,undefined)
00:36:08 WARN Bot finis() already quiting... return and wait...
ERROR: Bot exited with code 1
```

**Comment by windmemory** [comment]
Try to run with `v0.0.10`, might fix this issue :)

**Comment by kis87988** [comment]
I upgrade to `v0.0.10`, but it looks still have some problem.
I update the follow up log.
```log
15:35:04 SILL GrpcGateway parse(GrpcGetContact) get response: [{"MsgType":2,"UserName":"wxid_mask","NickName":"CTIOne训练营助手","Signature":"","SmallHeadImgUrl":"http://wx.qlogo.cn/mmhead/PiajxSqBRaEKibOFOdhSQ2OH4nibYWvHCTiala92fgT0mSkQ58YPFqKl0w/0","BigHe
15:35:04 SILL DedupeApi dedupeApi(GrpcGetContact) got results from external service, resolve 0 duplicate api calls.
15:35:04 SILL PadproGrpc GrpcGetContactPayload(wxid_mask)
15:35:04 SILL DedupeApi dedupeApi(GrpcGetContact, {"UserNameList":"wxid_mask"})
15:35:04 SILL DedupeApi dedupeApi(GrpcGetContact) deduped api call missed, call the external service.
15:35:04 SILL WechatGateway sendShort() res: commandUrl: /cgi-bin/micromsg-bin/getcontact
15:35:05 SILL GrpcGateway parse(GrpcGetContact) get response: [{"MsgType":2,"UserName":"wxid_mask","NickName":"YLF","Signature":"","SmallHeadImgUrl":"http://wx.qlogo.cn/mmhead/ver_1/FroVvicP8xegXmZH1YXF32lX6mjIxeeKlC9stPK5JO7tibGM6yKFVmTXk1QGYtCBKYCTpC
15:35:05 SILL DedupeApi dedupeApi(GrpcGetContact) got results from external service, resolve 0 duplicate api calls.
15:35:05 SILL PadproGrpc GrpcGetContactPayload(wxid_mask)
15:35:05 SILL DedupeApi dedupeApi(GrpcGetContact, {"UserNameList":"wxid_mask"})
15:35:05 SILL DedupeApi dedupeApi(GrpcGetContact) deduped api call missed, call the external service.
15:35:05 SILL WechatGateway sendShort() res: commandUrl: /cgi-bin/micromsg-bin/getcontact
15:35:06 SILL GrpcGateway parse(GrpcGetContact) get response: [{"MsgType":2,"UserName":"wxid_mask","NickName":"喵速","Signature":"𓆝𓆟𓆜𓆞𓆡𓆝𓆟𓆜𓆞𓆡","SmallHeadImgUrl":"h
ttp://wx.qlogo.cn/mmhead/icF4iau8Sj7b3ECUlNRyQQgGRkamOafbib5pbjZrAdKjC0VK4JkwibAbm
15:35:06 SILL DedupeApi dedupeApi(GrpcGetContact) got results from external service, resolve 0 duplicate api calls.
15:35:06 SILL PadproGrpc GrpcGetContactPayload(wxid_mask)
15:35:06 SILL DedupeApi dedupeApi(GrpcGetContact, {"UserNameList":"wxid_mask"})
15:35:06 SILL DedupeApi dedupeApi(GrpcGetContact) deduped api call missed, call the external service.
15:35:06 SILL WechatGateway sendShort() res: commandUrl: /cgi-bin/micromsg-bin/getcontact
15:35:07 SILL GrpcGateway parse(GrpcGetContact) get response: [{"MsgType":2,"UserName":"wxid_mask","NickName":"✨OLIVIA","Signature":"","SmallHeadImgUrl":"http://wx.qlogo.cn/mmhead/ ver_1/KNiaEBoUzUnRr7Mlfa3meRib9VHtOKDVIUbhjUeQcmqyaicP13KKC9XdJYxDfFuyLreQeUgBkWr
15:35:07 SILL DedupeApi dedupeApi(GrpcGetContact) got results from external service, resolve 0 duplicate api calls.
15:35:07 SILL PadproGrpc GrpcGetContactPayload(wxid_mask)
15:35:07 SILL DedupeApi dedupeApi(GrpcGetContact, {"UserNameList":"wxid_mask"})
15:35:07 SILL DedupeApi dedupeApi(GrpcGetContact) deduped api call missed, call the external service.
15:35:07 SILL WechatGateway sendShort() res: commandUrl: /cgi-bin/micromsg-bin/getcontact
15:35:07 SILL GrpcGateway parse(GrpcGetContact) get response: [{"MsgType":2,"UserName":"wxid_mask","NickName":"🍐梨芳如","Signature":"social phobia","SmallHeadImgUrl":"http
://wx.qlogo.cn/mmhead/ver_1/XbDsLJQqTeACLmYXTZjicfODpiax8XeDyFKZkoia58c43U5Nx3mG0zK
15:35:07 SILL DedupeApi dedupeApi(GrpcGetContact) got results from external service, resolve 0 duplicate api calls.
15:35:07 SILL PadproGrpc GrpcGetContactPayload(wxid_mask)
15:35:07 SILL DedupeApi dedupeApi(GrpcGetContact, {"UserNameList":"wxid_mask"})
15:35:07 SILL DedupeApi dedupeApi(GrpcGetContact) deduped api call missed, call the external service.
15:35:08 SILL WechatGateway sendShort() res: commandUrl: /cgi-bin/micromsg-bin/getcontact
15:35:17 SILL PadproManager debounceQueue heartbeat result: {}
15:35:17 SILL WechatGateway sendLong() reqSeq: 1410980424
15:35:17 SILL PadproManager throttleQueue emit heartbeat.
15:35:17 SILL PuppetPadpro startWatchdog() padproManager.on(heartbeat)
15:35:17 SILL Wechaty memoryCheck() free: 267 MB, require: 4 MB
15:35:17 SILL WechatGateway sendLong() receive back package size: 20
15:35:37 SILL PadproManager debounceQueue heartbeat result: {}
15:35:38 SILL WechatGateway sendLong() reqSeq: 770551614
15:35:38 SILL PadproManager throttleQueue emit heartbeat.
15:35:38 SILL PuppetPadpro startWatchdog() padproManager.on(heartbeat)
15:35:38 SILL Wechaty memoryCheck() free: 312 MB, require: 4 MB
15:35:38 SILL WechatGateway sendLong() receive back package size: 20
15:35:58 SILL PadproManager debounceQueue heartbeat result: {}
15:35:58 SILL WechatGateway sendLong() reqSeq: 1971679812
15:35:58 SILL PadproManager throttleQueue emit heartbeat.
15:35:58 SILL PuppetPadpro startWatchdog() padproManager.on(heartbeat)
15:35:58 SILL Wechaty memoryCheck() free: 312 MB, require: 4 MB
15:35:58 SILL WechatGateway sendLong() receive back package size: 20
15:36:18 SILL PadproManager debounceQueue heartbeat result: {}
15:36:19 SILL WechatGateway sendLong() reqSeq: 993706324
15:36:19 SILL PadproManager throttleQueue emit heartbeat.
15:36:19 SILL PuppetPadpro startWatchdog() padproManager.on(heartbeat)
15:36:19 SILL Wechaty memoryCheck() free: 304 MB, require: 4 MB
15:36:19 SILL WechatGateway sendLong() receive back package size: 20
15:36:39 SILL PadproManager debounceQueue heartbeat result: {}
15:36:39 SILL WechatGateway sendLong() reqSeq: 497719086
15:36:39 SILL PadproManager throttleQueue emit heartbeat.
15:36:39 SILL PuppetPadpro startWatchdog() padproManager.on(heartbeat)
15:36:39 SILL Wechaty memoryCheck() free: 304 MB, require: 4 MB
15:36:39 SILL WechatGateway sendLong() receive back package size: 20
15:36:59 SILL PadproManager debounceQueue heartbeat result: {}
15:37:00 SILL WechatGateway sendLong() reqSeq: 2035377403
15:37:00 SILL PadproManager throttleQueue emit heartbeat.
15:37:00 SILL PuppetPadpro startWatchdog() padproManager.on(heartbeat)
15:37:00 SILL Wechaty memoryCheck() free: 305 MB, require: 4 MB
15:37:00 SILL WechatGateway sendLong() receive back package size: 20
15:37:05 SILL WechatGateway sendLong() reqSeq: 1431988690
15:37:07 SILL WechatGateway sendLong() receive back package size: 30072
15:37:08 SILL GrpcGateway parse(GrpcSyncMessage) get response: [{"MsgType":2,"UserName":"wxid_mask","NickName":"呃...","Signature":"加冰加倍的快乐","SmallHeadImgUrl":"htt
p://wx.qlogo.cn/mmhead/DmTSLTdleesmxrEtxT4RzGjMz2nTM6icHeo6mj0h8Vnc3gA6Z9NRILg/0","BigHead
15:37:08 VERB PadproManager syncMessage() got 83 message(s).
{"MsgId":1467679367,"FromUserName":"wxid_mask","ToUserName":"wxid_b19nbi9oqkpg12","MsgType":1,"Content":"test","Status":3,"ImgStatus":1,"ImgBuf":null,"CreateTime":1543448225,"MsgSource":"<msgsource />\n","PushContent":"SU-William Chen    𓆜𓆝𓆞𓆡(wxid_mask) : test","NewMsgId":3458956744247730000}
15:37:08 VERB PuppetPadpro onPadproMessage({id=1467679367, type=Text(1)})
15:37:08 VERB Message static load(1467679367)
15:37:08 SILL Accessory #304<Message> constructor()
15:37:08 VERB Message constructor(1467679367) for class Message
15:37:08 VERB Message ready()
15:37:08 VERB Puppet messagePayload(1467679367)
15:37:08 SILL Puppet messagePayloadCache(1467679367) cache MISS
15:37:08 VERB PuppetPadpro messageRawPayloadParser({messageId="1467679367"})
15:37:08 SILL PuppetPadpro messagePayload({"id":"1467679367","timestamp":1543448225,"type":7,"fromId":"a38372624","text":"test","toId":"wxid_b19nbi9oqkpg12"})
15:37:08 SILL Puppet messagePayload(1467679367) cache SET
15:37:08 SILL Contact ready() @ Puppet#0<PuppetPadpro>(Ponyo_2)/PuppetPadpro#0
15:37:08 SILL Contact ready() isReady() true
15:37:08 SILL Contact ready() @ Puppet#0<PuppetPadpro>(Ponyo_2)/PuppetPadpro#0
15:37:08 SILL Contact ready() isReady() true
15:37:08 SILL Contact alias()
15:37:08 SILL Contact alias()
15:37:08 VERB Puppet selfId()
15:37:08 VERB Puppet selfId()
15:37:08 VERB Contact friend()
15:37:08 SILL Contact alias()
15:37:08 VERB Contact friend()
15:37:08 SILL Contact alias()
15:37:08 SILL Contact alias()
15:37:08 VERB Message say(I am still alive)
15:37:08 VERB PuppetPadpro messageSend({"contactId":"wxid_mask"}, I am still alive)
15:37:08 SILL PadproGrpc GrpcSendMessage(a38372624, I am still alive, undefined)
15:37:08 SILL WechatGateway sendShort() res: commandUrl: /cgi-bin/micromsg-bin/newsendmsg
15:37:09 SILL GrpcGateway parse(GrpcSendMessage) get response: null
15:37:14 SILL WechatGateway sendLong() reqSeq: 2126914130
15:37:15 INFO Bot finis(99,uncaughtException,Error: connect ETIMEDOUT 203.205.129.101:80)
15:37:15 INFO Bot finis() broadcast quiting message for bot
15:37:15 VERB Wechaty say(Wechaty will exit 99 because of uncaughtException )
15:37:15 VERB Puppet selfId()
15:37:15 VERB Contact say(Wechaty will exit 99 because of uncaughtException )
15:37:15 VERB PuppetPadpro messageSend({"contactId":"wxid_mask"}, Wechaty will exit 99 because of uncaughtException )
15:37:15 SILL PadproGrpc GrpcSendMessage(wxid_b19nbi9oqkpg12, Wechaty will exit 99 because of uncaughtException , undefined)
15:37:16 SILL WechatGateway sendLong() receive back package size: 30312
Error: connect ETIMEDOUT 203.205.129.101:80
    at TCPConnectWrap.afterConnect [as oncomplete] (net.js:1113:14)
15:37:16 INFO Bot finis(1,exit,undefined)
15:37:16 WARN Bot finis() already quiting... return and wait...
ERROR: Bot exited with code 1
  ____  _   _  ____   ____  _____ ____   ___  ____ _____
 | __ )| | | |/ ___| |  _ \| ____|  _ \ / _ \|  _ \_   _|
 |  _ \| | | | |  _  | |_) |  _| | |_) | | | | |_) || |
 | |_) | |_| | |_| | |  _ <| |___|  __/| |_| |  _ < | |
 |____/ \___/ \____| |_| \_\_____|_|    \___/|_| \_\|_|

Press ENTER to continue ... 
```

**Comment by windmemory** [comment]
This issue should be fixed with `v0.0.15`, please test this with that version.
